luigi/file.py

```diff
@@ -95,17 +148,19 @@ def makedirs(self):
         if parentfolder and not os.path.exists(parentfolder):
             os.makedirs(parentfolder)
 
-    def open(self, mode='r'):
+    def open(self, mode='r', timeout=10):
         if mode == 'w':
             self.makedirs()
             return self.format.pipe_writer(atomic_file(self.path))
-
+        elif mode == 'a':
+            self.makedirs()
+            return self.format.pipe_writer(atomic_file_append(self.path, timeout=timeout))
         elif mode == 'r':
             fileobj = FileWrapper(io.BufferedReader(io.FileIO(self.path, 'r')))
             return self.format.pipe_reader(fileobj)
 
         else:
-            raise Exception('mode must be r/w')
+            raise Exception('mode must be r, w or a')
```

Comment:
Or ArgumentError or whatever is most suitable. I don't know all exception types on top of my head :)